internal/consumer/partitionConsumer.go

```diff
@@ -238,7 +238,7 @@ func (p *partitionConsumer) messageLoop(offsetRange *kafka.OffsetRange) {
 				continue
 			}
 			drain = false
-
+			p.delayMsg(m.Timestamp)
```

Comment:
looks like we can simply pass `lag` instead of `m.Timestamp`, also we might want to some sanity check of `m.Timestamp`

Reply:
I think passing timestamp is a good idea because you can encapsulate validation and delay functionality in the `delayMsg` function.

Another alternative here is to make `delayMsg` only compute delay duration and not block. Blocking can happen explicitly here. e.g.,
```
time.Sleep(p.computeMsgDelay(m.Timestamp))

// No blocking
func computeMsgDelay(ts time.Time) time.Duration {...}
```

Reply:
if a real delay happens, lag calculated before delayMsg will be old and can not be used in the following metric. Thus, I put the lag calculation after p.delayMsg